lib/pure/options.nim

```diff
@@ -39,6 +39,29 @@ runnableExamples:
 ## inherits from `system.Defect` and should therefore never be caught.
 ## Instead, rely on checking if the option contains a value with the
 ## `isSome <#isSome,Option[T]>`_ and `isNone <#isNone,Option[T]>`_ procs.
+##
+##
+## Pattern matching
+## ================
+##
+## **Note:** This requires the [fusion](https://github.com/nim-lang/fusion) package.
+##
+## [fusion/matching](https://nim-lang.github.io/fusion/src/fusion/matching.html)
+## supports pattern matching on `Option`s, with the `Some(<pattern>)` and
+## `None()` patterns.
+##
+## .. code-block:: nim
+##   {.experimental: "caseStmtMacros".}
+##
+##   import fusion/matching
+##
+##   case some(42)
+##   of Some(@a):
+##     assert a == 42
+##   of None():
+##     assert false
+##
+##   Some(Some(None())) := some(some(none(int)))
```

Comment:
although some/Some/none/None is clear from the context, `:=` is not; can you add a short explanation?

Reply:
I could use `assertMatch` instead, which `:=` is an alias for.